pinot-spi/src/main/java/org/apache/pinot/spi/data/TimeFieldSpec.java

```diff
@@ -38,65 +38,6 @@ public TimeFieldSpec() {
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, TimeUnit incomingTimeUnit, String outgoingName,
-      DataType outgoingDataType, TimeUnit outgoingTimeUnit) {
-    super(outgoingName, outgoingDataType, true);
-    _incomingGranularitySpec = new TimeGranularitySpec(incomingDataType, incomingTimeUnit, incomingName);
-    _outgoingGranularitySpec = new TimeGranularitySpec(outgoingDataType, outgoingTimeUnit, outgoingName);
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, TimeUnit incomingTimeUnit, String outgoingName,
-      DataType outgoingDataType, TimeUnit outgoingTimeUnit, Object defaultNullValue) {
-    super(outgoingName, outgoingDataType, true, defaultNullValue);
-    _incomingGranularitySpec = new TimeGranularitySpec(incomingDataType, incomingTimeUnit, incomingName);
-    _outgoingGranularitySpec = new TimeGranularitySpec(outgoingDataType, outgoingTimeUnit, outgoingName);
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, int incomingTimeUnitSize,
-      TimeUnit incomingTimeUnit) {
-    super(incomingName, incomingDataType, true);
-    _incomingGranularitySpec =
-        new TimeGranularitySpec(incomingDataType, incomingTimeUnitSize, incomingTimeUnit, incomingName);
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, int incomingTimeUnitSize,
-      TimeUnit incomingTimeUnit, Object defaultNullValue) {
-    super(incomingName, incomingDataType, true, defaultNullValue);
-    _incomingGranularitySpec =
-        new TimeGranularitySpec(incomingDataType, incomingTimeUnitSize, incomingTimeUnit, incomingName);
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, int incomingTimeUnitSize,
-      TimeUnit incomingTimeUnit, String outgoingName, DataType outgoingDataType, int outgoingTimeUnitSize,
-      TimeUnit outgoingTimeUnit) {
-    super(outgoingName, outgoingDataType, true);
-    _incomingGranularitySpec =
-        new TimeGranularitySpec(incomingDataType, incomingTimeUnitSize, incomingTimeUnit, incomingName);
-    _outgoingGranularitySpec =
-        new TimeGranularitySpec(outgoingDataType, outgoingTimeUnitSize, outgoingTimeUnit, outgoingName);
-  }
-
-  public TimeFieldSpec(String incomingName, DataType incomingDataType, int incomingTimeUnitSize,
-      TimeUnit incomingTimeUnit, String outgoingName, DataType outgoingDataType, int outgoingTimeUnitSize,
-      TimeUnit outgoingTimeUnit, Object defaultNullValue) {
-    super(outgoingName, outgoingDataType, true, defaultNullValue);
-    _incomingGranularitySpec =
-        new TimeGranularitySpec(incomingDataType, incomingTimeUnitSize, incomingTimeUnit, incomingName);
-    _outgoingGranularitySpec =
-        new TimeGranularitySpec(outgoingDataType, outgoingTimeUnitSize, outgoingTimeUnit, outgoingName);
-  }
-
   public TimeFieldSpec(TimeGranularitySpec incomingGranularitySpec) {
```

Comment:
We don't allow null for time, so you can remove the APIs with `defaultNullValue` as well

Reply:
done